app/src/main/java/com/babylon/wallet/android/domain/model/IncomingMessage.kt

```diff
@@ -12,41 +13,64 @@ import com.radixdlt.sargon.IdentityAddress
 import com.radixdlt.sargon.LedgerHardwareWalletModel
 import com.radixdlt.sargon.NetworkId
 import com.radixdlt.sargon.RequestedNumberQuantifier
-import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.WalletInteractionId
 import kotlinx.parcelize.Parcelize
 import kotlinx.serialization.Serializable
 import rdx.works.core.domain.TransactionManifestData
 import rdx.works.core.sargon.PersonaDataField
 
-sealed interface MessageFromDataChannel {
+sealed interface IncomingMessage {
+
+    sealed interface RemoteEntityID {
+
+        val value: String
+
+        data class RadixMobileConnectRemoteSession(val id: String) : RemoteEntityID {
+            override val value: String
+                get() = id
+        }
+
+        data class ConnectorId(val id: String) : RemoteEntityID {
+            override val value: String
+                get() = id
+        }
+    }
 
     sealed class IncomingRequest(
-        open val remoteConnectorId: String, // from which remote CE comes the message
-        val id: String, // the id of the request
+        open val remoteEntityId: RemoteEntityID, // from which remote source message came
+        open val interactionId: WalletInteractionId, // the id of the request
         val metadata: RequestMetadata
-    ) : MessageFromDataChannel {
+    ) : IncomingMessage {
 
-        val isInternal: Boolean
+        open val isInternal: Boolean
```

Comment:
Very nice that you did split the incoming messages with the notion of `RemoteEntityId`. On the other hand, the `isInternal` is becoming now confusing. I will see what I can do when I will work on this PR. We could explore maybe renaming the `RemoteEntityID` to `RequestEntity`. This entity can be
* for radix connect
* for CE
* for internal messages.

I have not yet though about it throughout, but I think it is worth exploring.